Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -195,66 +195,50 @@
 	end
 end
 
-
 %% 3. Calculate the number of repetitions
-
-% #997 -> Jan I tried to rewrite it, so it becomes fully understandable.
-% Why don't we actually use the names nBlocks and nVolumesPerBlock
-% isn't that much less confusion-prone? Because "nFullHadamardVolumes" to me could mean different things
-
 % Here we define the terminology about repetitions and blocks
 % nEncodedVolumes = total original volumes
 %
-%   A "block" is in mathematics a submatrix, a section of a matrix
-%   With Hadamard-encoded ASL, the fully encoded Hadamard matrix is separated in e.g., 4 or 8 blocks (corresponding to Hadamard-4 or Hadamard-8, with
-%   1 control volume and 3 or 7 PLDs, respectively). So:
+% With Hadamard-encoded ASL, the fully encoded Hadamard matrix is separated in several (e.g., 4 or 8) blocks corresponding to Hadamard-4 or Hadamard-8.
+% This means that before each acquisition we alternate 4 or 8 blocks of control/label as specified by the Hadamard matrix.
+% By block-wise subtraction and addition of the 4 or 8 acquired volumes according to the Hadamard encoding matrix, 
+% we can decode 1 control volume and 3 or 7 PLDs, respectively.
 %
-% block = control or single PLD acquisition encoded in Hadamard
 % TimeEncodedMatrixSize = nBlocks
 
 nBlocks = xQ.TimeEncodedMatrixSize;
 
-%   INNER REPETITIONS
-%   There may be readout-related reasons for having multiple volumes.
-%   E.g., with a multi-TE readout, each control or label volume (in this case each Hadamard-encoded volume) is acquired multiple 
-%   times for the number of echoes. Because this is repeated at the readout level, we define these here as "inner repetitions".
-%   E.g., for a 8 echoes Hadamard acquisition, a single control or PLD-encoded acquisition is repeated 8 times, so:
+% INNER REPETITIONS
+% Normally, for a 4-block Hadamard, we acquire 4 volumes. But there may be readout-related reasons for having more volumes than that.
+% E.g., with a multi-TE readout, each Hadamard-encoded volume is acquired multiple times equal to the number of echoes. Because this is repeated 
+% at the readout level, we define these here as "inner repetitions". E.g., for a 8-echoes Hadamard acquisition, a single control or PLD-encoded acquisition is repeated 8 times.
 %
 % innerRepetition: single-volume repetitions within a Hadamard block
 
 [~, ~, indexUniquePLD] = unique(xQ.Initial_PLD); % Find volume indices of unique PLDs
-indexFirstPLD = find(indexUniquePLD==1); % Find the volumes of the first PLD
-indexSecondPLD = find(indexUniquePLD==2); % Find the volumes of the second PLD
+indexSecondPLD = find(indexUniquePLD==2); % Find the volumes of the second PLD - all PLDs before the second PLD are equivalent
```

Comment:
Yes exactly. 1000 first, 1500 second in your example. But in case somebody puts 1000, 1000, 1500, 1500, then the inner repetition == 2 and we detect that by having the first index of second PLD on position 3 instead of the position 2.

I've clarified it in the code.